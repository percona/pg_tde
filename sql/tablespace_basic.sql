\set tde_am tde_heap_basic
\i sql/tablespace.inc