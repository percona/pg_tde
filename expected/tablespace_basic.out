\set tde_am tde_heap_basic
\i sql/tablespace.inc
CREATE EXTENSION pg_tde;
SELECT pg_tde_add_key_provider_file('file-vault','/tmp/pg_tde_test_keyring.per');
 pg_tde_add_key_provider_file 
------------------------------
                            1
(1 row)

SELECT pg_tde_set_principal_key('test-db-principal-key','file-vault');
 pg_tde_set_principal_key 
--------------------------
 t
(1 row)

CREATE TABLE test(num1 bigint, num2 double precision, t text) USING :tde_am;
INSERT INTO test(num1, num2, t)
  SELECT round(random()*100), random(), 'text'
  FROM generate_series(1, 10) s(i);
CREATE INDEX test_idx ON test(num1);
SET allow_in_place_tablespaces = true;
CREATE TABLESPACE test_tblspace LOCATION '';
ALTER TABLE test SET TABLESPACE test_tblspace;
SELECT count(*) FROM test;
 count 
-------
    10
(1 row)

ALTER TABLE test SET TABLESPACE pg_default;
REINDEX (TABLESPACE test_tblspace, CONCURRENTLY) TABLE test;
INSERT INTO test VALUES (110, 2);
SELECT * FROM test WHERE num1=110;
 num1 | num2 | t 
------+------+---
  110 |    2 | 
(1 row)

DROP TABLE test;
DROP TABLESPACE test_tblspace;
DROP EXTENSION pg_tde;
